sql/index/pilosa/driver.go

```diff
@@ -65,85 +75,95 @@ func (*Driver) ID() string {
 
 // Create a new index.
 func (d *Driver) Create(db, table, id string, expr []sql.ExpressionHash, config map[string]string) (sql.Index, error) {
-	path, err := mkdir(d.root, db, table, id)
+	_, err := mkdir(d.root, db, table)
 	if err != nil {
 		return nil, err
 	}
 
+	if config == nil {
+		config = make(map[string]string)
+	}
+
+	config["index"] = indexName(db, table)
+	for _, e := range expr {
+		config[hex.EncodeToString(e)] = frameName(id, e)
+	}
 	cfg := index.NewConfig(db, table, id, expr, d.ID(), config)
-	err = index.WriteConfigFile(path, cfg)
+	err = index.WriteConfigFile(d.configFileName(db, table, id), cfg)
 	if err != nil {
 		return nil, err
 	}
 
-	return newPilosaIndex(path, d.client, cfg), nil
+	return newPilosaIndex(d.mappingFileName(db, table, id), d.client, cfg), nil
 }
 
 // LoadAll loads all indexes for given db and table
 func (d *Driver) LoadAll(db, table string) ([]sql.Index, error) {
-	root := filepath.Join(d.root, db, table)
+	path := filepath.Join(d.root, db, table)
 
 	var (
 		indexes []sql.Index
 		errors  []string
-		err     error
 	)
-	filepath.Walk(root, func(path string, info os.FileInfo, err error) error {
-		if err != nil {
-			if path != root || !os.IsNotExist(err) {
-				errors = append(errors, err.Error())
-			}
-			return filepath.SkipDir
-		}
 
-		if info.IsDir() && path != root && info.Name() != "." && info.Name() != ".." {
-			idx, err := d.loadIndex(path)
-			if err != nil {
-				if !errCorruptedIndex.Is(err) {
-					errors = append(errors, err.Error())
-				}
+	configfiles, err := filepath.Glob(filepath.Join(path, "*") + ConfigFileExt)
+	if err != nil {
+		return nil, err
+	}
 
-				return filepath.SkipDir
+	for _, config := range configfiles {
+		idx, err := d.loadIndex(config)
+		if err != nil {
+			if !errCorruptedIndex.Is(err) {
+				errors = append(errors, err.Error())
 			}
 
-			indexes = append(indexes, idx)
+			continue
 		}
 
-		return nil
-	})
-
+		indexes = append(indexes, idx)
+	}
 	if len(errors) > 0 {
-		err = fmt.Errorf(strings.Join(errors, "\n"))
+		return nil, fmt.Errorf(strings.Join(errors, "\n"))
 	}
-	return indexes, err
+	return indexes, nil
 }
 
-func (d *Driver) loadIndex(path string) (sql.Index, error) {
-	ok, err := index.ExistsProcessingFile(path)
+func (d *Driver) loadIndex(configfile string) (sql.Index, error) {
+	mapping := strings.Replace(configfile, ConfigFileExt, MappingFileExt, 1)
+	processing := strings.Replace(configfile, ConfigFileExt, ProcessingFileExt, 1)
+	ok, err := index.ExistsProcessingFile(processing)
 	if err != nil {
 		return nil, err
 	}
 
 	if ok {
 		log := logrus.WithFields(logrus.Fields{
 			"err":  err,
-			"path": path,
+			"path": configfile,
 		})
 		log.Warn("could not read index file, index is corrupt and will be deleted")
 
-		if err := os.RemoveAll(path); err != nil {
-			log.Warn("unable to remove folder of corrupted index")
+		if err := os.RemoveAll(configfile); err != nil {
```

Comment:
I would prefer to delete the path instead of each file separately. Why we change it?

Reply:
The problem here is folder's structure. We don't have `id` subfolder anymore. Just all `db/table` indexes go to the one `pilosa index` folder.
I wanted to keep the `id` folder for resources like config, mapping, etc. but it will distract `pilosa/pilosa`, because when you load an index in pilosa  it goes through subfolders (and don't validate it), so at the end it creates fake/invalid extra indexes. That's why I flatten the structure and left subfolders to pilosa internals.

But right now I thought about another folders refactoring.
How about:
```bash
root
|-db
   |-table
      |-[driver id] # pilosa folder if needed
      |-[id 1] # index metadata (config, mapping, ...)
      |-[id 2]
...
```
The pros. here is the structure is almost what we had before, so deleting the index is easier - you can just delete a `id` folder.
The cons. yo don't have connection between `id` and `driver`. If you have multiple indexes on the same table from different drivers you don't know which index belongs to which driver. What in theory may not be problem if we mark it in a config file.
WDYT?